### PR TITLE
Add Java templates

### PR DESCRIPTION
**Pull request by adimiz1**
1. Add `README.md` template  for Java
2. Update `Configruation.mustache` to consume Cloudinary url
3. Add `config.yml` for Java with the required for fields for branding and module names
4. Update folder name to `media_delivery_java`

### CODE REVIEW

File: templates/java/Configuration.mustache [hunk 1]
```diff
@@ -0,0 +1,102 @@
...
+
+    static private void updateClient(Map config) {
+        HttpBasicAuth basicAuth = (HttpBasicAuth) defaultApiClient.getAuthentication("basicAuth");
+        basicAuth.setUsername((String) config.get("api_key"));
+        basicAuth.setPassword((String) config.get("api_secret"));
+        buildCloudinaryUrl((String) config.get("cloud_name"));
+    }
+
+    static private void buildCloudinaryUrl(String cloudName) {
+        defaultApiClient.setBasePath("https://api-staging.cloudinary.com/v2/" + cloudName);
```
Comment (const-cloudinary) [inline templates/java/Configuration.mustache]: Having a base path hardcoded is not a good idea, we get it from the YML file.
And in this case it points to `staging` which is also not production ready

**Review by const-cloudinary — CHANGES_REQUESTED** [review]

File: templates/java/Configuration.mustache [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+    }
+
+    static private void updateClient(Map config) {
+        HttpBasicAuth basicAuth = (HttpBasicAuth) defaultApiClient.getAuthentication("basicAuth");
+        basicAuth.setUsername((String) config.get("api_key"));
+        basicAuth.setPassword((String) config.get("api_secret"));
+        buildCloudinaryUrl((String) config.get("cloud_name"));
+    }
+
+    static private void buildCloudinaryUrl(String cloudName) {
```
Comment (const-cloudinary) [inline templates/java/Configuration.mustache]: and the function name is a bit confusing, it does not build the CLOUDINARY_URL, but updates basePath with the cloud name that was parsed from the CLOUDINARY_URL

File: templates/java/Configuration.mustache (hunk 1 above)
Comment (adimiz1) [reply templates/java/Configuration.mustache]: I'll look it up in the YML and change it

File: templates/java/Configuration.mustache (hunk 2 above)
Comment (adimiz1) [reply templates/java/Configuration.mustache]: Fixed

File: configs/java/config.yml [hunk 3]
```diff
@@ -0,0 +1,11 @@
+additionalProperties:
+  modelPackage: com.cloudinary.model
+  appVersion: 1.0.0
```
Comment (strausr) [inline configs/java/config.yml]: Shouldn't the app version be dynamic?

Comment (adimiz1) [reply configs/java/config.yml]: Removed

**Review by strausr — APPROVED** [review]

**Review by const-cloudinary — APPROVED** [review]

**Pull request merged**